x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/notifications_index_mappings.json

```diff
@@ -27,6 +27,9 @@
     },
     "job_type": {
       "type": "keyword"
+    },
+    "cleared": {
```

Comment:
Just curious: Is it the first time or did it happen in the past that we added mappings for fields used solely by the UI?

Reply:
I think this is the first case of a field that is never touched in any way by the backend.

There are other fields that the backend knows about but doesn't care about, like `custom_settings` on jobs and `modified_time`/`modified_username` on annotations.

I guess in the future we can extend the parser and xcontent generator on `AbstractAuditMessage` to preserve the `cleared` field through a parse and rewrite.  However, we don't need that at the moment and I am thinking that it might be best to wait to see how the UI change works out before adding code to the backend that might need to be changed in the future.

Reply:
Thanks for explanation.
For sure we need to remember (or better, document?) that this field must not be modified by the backend. Otherwise, we may have concurrent modification bugs.